### PR TITLE
Gemfile/Gemspec Tidy (For Master)

### PR DESCRIPTION
**Pull request by luke-hill**
## Summary

Tidy up Gemfile / Deps

## Details

Migrate everything that could be out of gemfile conditionals into gemspec fixed values.
NB: I've also updated a few minor version restrictions. And the *minimum ruby version* although this was always 2.3.3 due to a dependency requirement.

## Motivation and Context

Tidies things up.

## How Has This Been Tested?

CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (cleanup of codebase without changing any existing functionality)
- [ ] Update documentation

## Checklist:

- [ ] I've added tests for my code
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by luke-hill** [comment]
I'll make one of these for `still` in a day or so.

**Comment by luke-hill** [comment]
ping @mvz i've no idea if the CI here is stable or not? Could you please advise? Can this be merged?

**Comment by mvz** [comment]
I think an adjustment to Travis' configuration is needed. Hold on.

File: aruba.gemspec [hunk 1]
```diff
@@ -21,9 +19,21 @@ Gem::Specification.new do |spec|
...
 
-  spec.add_development_dependency 'bundler', ['>= 1.7.0', '< 3.0']
+  spec.add_development_dependency 'fuubar', '~> 2.3'
+  spec.add_development_dependency 'json', '~> 2.1'
+  spec.add_development_dependency 'license_finder', '~> 5.3'
+  spec.add_development_dependency 'minitest', '~> 5.10'
+  spec.add_development_dependency 'pry-doc', '~> 1.0.0'
+  spec.add_development_dependency 'rake', '~> 12.3'
+  spec.add_development_dependency 'rspec', '~> 3.6'
+  spec.add_development_dependency 'rubocop', '~> 0.69'
```
Comment (mvz) [inline aruba.gemspec]: This should be `~> 0.69.0` because the set of cops can change with each release.

File: aruba.gemspec [hunk 2]
```diff
@@ -21,9 +19,21 @@ Gem::Specification.new do |spec|
...
+  spec.add_development_dependency 'rubocop-performance', '~> 1.1'
+  spec.add_development_dependency 'simplecov', '~> 0.15'
+  spec.add_development_dependency 'travis-yaml', '~> 0.2'
+  spec.add_development_dependency 'yard-junk', '~> 0.0.7'
+
   spec.rubygems_version = '>= 1.6.1'
-  spec.required_ruby_version = '>= 2.3'
+  spec.required_ruby_version = '>= 2.3.3'
```
Comment (mvz) [inline aruba.gemspec]: I'm not a fan of excluding micro versions in general. Can we avoid this? Which of our depedencies needs 2.3.3? Maybe we need to test on 2.3.0 in Travis as well?

**Comment by mvz** [comment]
The build should be green now. The signs are good :slightly_smiling_face:.

File: aruba.gemspec (hunk 1 above)
Comment (luke-hill) [reply aruba.gemspec]: I agree with this, but I just copied the existing config, all this PR is aimed to be is a tidy up.

File: aruba.gemspec (hunk 2 above)
Comment (luke-hill) [reply aruba.gemspec]: One of the documentation ones in the development dependencies, if we reduce it down we should be ok

**Pull request merged**